### PR TITLE
Count filters in addition to variables

### PR DESCRIPTION
**Pull request by brian-vanderford**
Makes Template.analyze and friends count filters in addition to variables.

### CODE REVIEW

File: tests/test_analyze_template.py [hunk 1]
```diff
@@ -95,6 +96,8 @@ def _test(
         self.assertEqual(refs.variables, template_refs)
         self.assertEqual(refs.failed_visits, failed_visits)
         self.assertEqual(refs.unloadable_partials, unloadable)
+        if template_filters:
```
Comment (jg-rp) [inline tests/test_analyze_template.py:99]: I'd like to assert that `refs.filters` an empty dictionary in the event that the `template_filters` argument is `None`. Mainly to catch any accidental omissions of `template_filters` from future test cases.

File: tests/test_contextual_analysis.py [hunk 2]
```diff
@@ -37,6 +40,8 @@ async def coro():
         self.assertEqual(refs.all_variables, variables)
         self.assertEqual(refs.local_variables, assigns)
         self.assertEqual(refs.undefined_variables, undefined)
+        if filters is not None:
```
Comment (jg-rp) [inline tests/test_contextual_analysis.py:43]: Same as previous comment. Assert that `refs.filters` is empty if not given.

```python
else:
    self.assertEqual(refs.filters, {})
```

**Review by jg-rp — CHANGES_REQUESTED** [review]
Hi @brian-vanderford,

Looks good. Would counting tags be a useful feature too?

File: tests/test_analyze_template.py (hunk 1 above)
Comment (brian-vanderford) [reply tests/test_analyze_template.py:99]: Done in

File: tests/test_contextual_analysis.py (hunk 2 above)
Comment (brian-vanderford) [reply tests/test_contextual_analysis.py:45]: Done

**Comment by brian-vanderford** [comment]
> Looks good. Would counting tags be a useful feature too?

It didn't occur to me, but I could see the same use case as with filters; you can't render the template, but you'd still like to tell your users when they're trying to use a tag that won't be defined when it does get rendered.

**Review by jg-rp — APPROVED** [review]

**Pull request merged**